### PR TITLE
[EAK-344] Added documentation and test for Slider component, added validations for slider widget fields.

### PR DESCRIPTION
**Pull request by gmusabaev**
[EAK-344] Added documentation and test for Slider component, added validations for slider widget fields.

### CODE REVIEW

File: plugin/src/test/resources/widget/slider/_cq_dialog.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                            sling:resourceType="etoolbox-authoring-kit/components/authoring/slider"
+                            name="./number"
+                            fieldLabel="Number"
+                            renderHidden="{Boolean}false"
+                            required="{Boolean}false"
+                            min="{Long}10"
+                            max="{Long}200"
+                            step="{Long}2"
+                            orientation="vertical"
+                            filled="{Boolean}true"
```
Comment (AlKaliada) [inline plugin/src/test/resources/widget/slider/_cq_dialog.xml:27]: oriental should be VERTICAL like in plugin/src/test/com/exadel/aem/toolkit/test/widget/SliderWidget.java and add disabled="{Boolean}false"

**Review by AlKaliada — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java [hunk 2]
```diff
@@ -29,15 +31,19 @@
...
+    @ValueRestriction(ValueRestrictions.NON_NEGATIVE)
     long min() default 0;
 
+    @ValueRestriction(ValueRestrictions.NON_NEGATIVE)
     long max() default 100;
 
+    @ValueRestriction(ValueRestrictions.POSITIVE)
     long step() default 1;
 
     Orientation orientation() default Orientation.HORIZONTAL;
```
Comment (AlKaliada) [inline core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java:43]: orientation doesn't work, pls override toString() in Orientation enum
@Override
    public String toString() {
        return name().toLowerCase();
    }

**Review by AlKaliada — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java (hunk 2 above)
Comment (AlKaliada) [reply core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java:43]: and maybe should add getter for request in BaseModel

Comment (smiakchilo) [reply core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java:43]: > orientation doesn't work, pls override toString() in Orientation enum

Good point! But we use  `@PropertyRendering(transform = StringTransformation.LOWERCASE)` meta-annotation. You see, there are cases when all the same enum once needed in caps, other time in lowerase

Comment (smiakchilo) [reply core/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/Slider.java:43]: > and maybe should add getter for request in BaseModel

👍

File: plugin/src/test/com/exadel/aem/toolkit/plugin/maven/WidgetsTest.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
  */
 package com.exadel.aem.toolkit.plugin.maven;
 
+import com.exadel.aem.toolkit.test.widget.SliderWidget;
```
Comment (smiakchilo) [inline plugin/src/test/com/exadel/aem/toolkit/plugin/maven/WidgetsTest.java:16]: (cosmetic) Please make sure that the `.editorconfig` file is recognized and honored by your Idea. `com.exadel.*` imports always go last

**Pull request closed without merging**